Apromore-Core-Components/Apromore-BPMNEditor/src/main/scripts/plugins/pdf.js

```diff
@@ -108,26 +108,33 @@ Apromore.Plugins.File = Clazz.extend({
         }
 
         // Send the svg to the server.
-        new Ajax.Request(Apromore.CONFIG.PDF_EXPORT_URL, {
-            method: 'POST',
-            parameters: {
-                resource: resource,
-                data: svgClone,
-                format: "pdf"
-            },
-            onSuccess: (function(request){
-                myMask.hide();
-                // Because the pdf may be opened in the same window as the
-                // process, yet the process may not have been saved, we're
-                // opening every other representation in a new window.
-                // location.href = request.responseText
-                window.open(request.responseText);
-            }).bind(this),
-            onFailure: (function(){
+        var xhr = new XMLHttpRequest();
+        var params = "resource=" + resource + "&data=" + svgClone + "&format=pdf";
+
+        xhr.open("POST", Apromore.CONFIG.PDF_EXPORT_URL);
+        xhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+        xhr.responseType = 'arraybuffer';
+        xhr.onreadystatechange = function() {
+            if (xhr.readyState == 4 && xhr.status == 200) {
+                // Download pdf from blob
                 myMask.hide();
-                Ext.Msg.alert(window.Apromore.I18N.Apromore.title, window.Apromore.I18N.File.genPDFFailed);
-            }).bind(this)
-        });
+                var resultByte = xhr.response;
+                var bytes = new Uint8Array(resultByte); // convert to byte array
+                var blob = new Blob([bytes], {type: "application/pdf"});// convert to a blob
+
+                var hiddenElement = document.createElement('a');
+                hiddenElement.href = window.URL.createObjectURL(blob);
+                hiddenElement.target = '_blank';
+                hiddenElement.download = 'diagram.pdf';

```

Comment:
Could we get model name?

Reply:
I'm having trouble figuring out how to get the model name in pdf.js. We can leave it as diagram.pdf for now and update with the model name at a later stage.